### PR TITLE
Turn remacs_sys into a module (rather than a crate)

### PR DESCRIPTION
**Pull request by pipcet**
This is related to #706. I'd like to merge `Lisp_Object` and `LispObject`. That appears to require turning `remacs_sys` into a module that's part of the `remacs` crate, but I think doing so is a good idea anyway, so this is an independent PR.

Keeping `remacs-sys` a separate crate means duplication, fewer warnings, and introduces a notion of independence that simply isn't appropriate.

`remacs-sys` is meant to shrink as code is translated, so it's not "Rust bindings for Emacs". Any attempt to build mismatched versions of `remacs` and `remacs-sys` is bound to fail, and that's a good argument for not keeping them in separate crates. As the `LispObject` case demonstrates, it's reasonable to want to extend Emacs C types using `remacs` functions; usually we're avoiding that by defining part of the C interface in `remacs` anyway, but that's just a lucky accident.

Rust analyzes unused functions, at least, on a crate basis, so we cannot hope to ever enable these warnings if we keep `remacs-sys` separate.

As far as I can make out, the obvious safety/isolation arguments aren't valid because Rust supports namespacing on a module level.

Again, it is my hope to combine the `Lisp*Object`s and that's my immediate reason for this PR, but I think it would be helpful to discuss this separately: even if you disagree that combining the `Lisp*Object`s is something remacs should do, it's something that we *should be able to do* easily, and that we're not is a sign something's wrong with our crate system.

If you're making the safety counterargument, I would appreciate it, if possible, if you could provide actual examples of code that becomes unsafe if the separation into crates is dropped. I know that dropping the separation would make code valid that currently isn't, but it doesn't follow that the newly-valid code is also unsafe.

### CODE REVIEW

File: rust_src/README.md [hunk 1]
```diff
@@ -1,10 +1,10 @@
 # remacs-sys
 
-`remacs-sys` is a Rust library containing all the [Foreign Function Interface (FFI)](https://doc.rust-lang.org/book/first-edition/ffi.html) declarations of structs and functions that are defined in C. Then, they can be used in the Rust code. C implementations are in `remacs/src`.
+`remacs-sys` is a Rust module containing all the [Foreign Function Interface (FFI)](https://doc.rust-lang.org/book/first-edition/ffi.html) declarations of structs and functions that are defined in C. Then, they can be used in the Rust code. C implementations are in `remacs/src`.
 
 ## Example: Use `make_float` C function in Rust
 
-First, let's have a look how `make_float` is defined historically in the C code. Then, how one can re-use it in Rust.
+First, let's have a look how `make_float` is defined historically in the C code. Then, how one can re-use imt in Rust.
```
Comment (shaleh) [inline rust_src/README.md]: Typo. `s/imt/it/`

File: rust_src/README.md [hunk 2]
```diff
@@ -48,10 +48,10 @@ use libc::{c_double, /* ... */ };
 // ...
 ```
 
-Second, `Lisp_Object` is an alias to `EmacsInt`:
+Second, `Lisp_Object` is a struct containing an EmacsInt:
```
Comment (shaleh) [inline rust_src/README.md]: It currently is an alias.

File: rust_src/README.md [hunk 3]
```diff
@@ -48,10 +48,10 @@ use libc::{c_double, /* ... */ };
...
 
 ```rust
-pub type Lisp_Object = EmacsInt;
+pub struct Lisp_Object(EmacsInt);
```
Comment (shaleh) [inline rust_src/README.md]: Not true yet.

File: rust_src/README.md (hunk 1 above)
Comment (pipcet) [reply rust_src/README.md]: Thanks

File: rust_src/README.md (hunk 2 above)
Comment (pipcet) [reply rust_src/README.md]: Hmm. Not in Wilfred/master, unless I'm missing something:
```
rust_src/remacs-sys/lib.rs:pub struct Lisp_Object(EmacsInt);
```

File: rust_src/README.md (hunk 3 above)
Comment (pipcet) [reply rust_src/README.md]: Again, that's what I'm seeing in a fresh checkout.

File: rust_src/README.md (hunk 2 above)
Comment (Wilfred) [reply rust_src/README.md]: As of #562 it's not an alias.

**Comment by Wilfred** [comment]
Getting better compiler warnings and simplifying our datatypes seems very worthwhile to me :+1: 

I'm open to getting rid of `emacs-sys` and `Lisp_Object` (I will try to review properly tomorrow). We've iterated on the design of Remacs internals, and I think the codebase is getting cleaner over time. Nonetheless, some design decisions don't work out. We don't have any epic safety arguments in favour of a separate crate, for example.

File: rust_src/README.md (hunk 2 above)
Comment (shaleh) [reply rust_src/README.md]: Yeah, my bad. That was even my PR too. Sigh. I have swapped out remacs for the work stack.

**Pull request merged**